### PR TITLE
Implement Doppler Secret Push and Delete functions

### PR DESCRIPTION
**Pull request by filedeploy**
## Problem Statement

Implement existing Doppler client push and delete functions.

I have manually tested both functions with KinD and an existing Doppler project.

## Related Issue

None

## Proposed Changes

N/A

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by filedeploy** [comment]
@ryan-blunden Feel free to give this PR a 👍/👎 as well.

File: pkg/provider/doppler/client.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ const (
 	verifyTLSOverrideEnvVar                            = "DOPPLER_VERIFY_TLS"
 	errGetSecret                                       = "could not get secret %s: %s"
 	errGetSecrets                                      = "could not get secrets %s"
+	errDeleteSecrets                                   = "could not delete secrets %s: %s"
+	errPushSecrets                                     = "could not push secrets %s: %s"
```
Comment (Skarlso) [inline pkg/provider/doppler/client.go:41]:
```suggestion
	errDeleteSecrets                                   = "could not delete secrets %s: %w"
	errPushSecrets                                     = "could not push secrets %s: %w"
```

Comment (filedeploy) [reply pkg/provider/doppler/client.go]: 🙏 I went ahead and replaced the `%s` with `%w` in two adjacent lines as well.

Comment (Skarlso) [reply pkg/provider/doppler/client.go]: Thanks!

**Review by Skarlso — APPROVED** [review]

**Pull request merged**